src/pam-ssh-add/pam-ssh-add.c

```diff
@@ -815,6 +815,7 @@ stash_password_for_session (pam_handle_t *pamh,
       return PAM_AUTHTOK_RECOVER_ERR;
     }
 
+  free_password (password_copy);
```

Comment:
It feels like I just touched that code.. commit ce1c6f91ecd1


I think this is wrong -- `pam_set_data` "consumes" the copy on success (i.e. it holds a reference to it and defines its lifetime), hence the need for passing the free function. Like this it's access to freed memory. See [manpage](https://linux.die.net/man/3/pam_set_data)

If that wasn't so, we would not need to bother with copying `password` in the first place. But `pam_set_data()` isn't in a position to copy the pointer itself, as it doesn't know its size or structure.

So this needs an annotation instead.

Reply:
> So this needs an annotation instead.

Annotation?

Reply:
Like in commit 9d48655bc3 , see e.g. [stackoverflow](https://stackoverflow.com/questions/69596188/how-can-i-disable-coverity-checking-using-code-annotation).